src/main/java/org/assertj/core/api/AbstractThrowableAssert.java

```diff
@@ -158,6 +158,30 @@ public SELF hasNoCause() {
     return myself;
   }
 
+  /**
+   * Verifies that the actual {@code Throwable} has a cause.
+   *
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code Throwable} is {@code null}.
+   * @throws AssertionError if the actual {@code Throwable} does not have a cause.
+   */
+  public SELF hasCause() {
```

Comment:
What if we add `getCause()` instead, which checks if there is a cause and returns a new `ThrowableAssert`? This would be symmetrical with `havingCause()`, which could become:

```java
  public ThrowableAssertAlternative<?> havingCause() {
    AbstractThrowableAssert<?, ?> causeAssert = delegate.getCause();
    return new ThrowableAssertAlternative<>(causeAssert.actual);
  }
```

Same would be for `getRootCause()` instead of `hasRootCause()`.

I propose this option also because `hasCause()` would have a slightly different purpose compared to the existing `hasCause(Throwable)`, i.e., checking the cause existence vs. comparison with the input value.

Reply:
Would it not be better to _add_ the `getCause()` method in `AbstractThrowableAssert` on top? In that case, it would be like this:

```
  public ThrowableAssert getCause() {
    hasCause();
    return new ThrowableAssert(actual.getCause());
  }
```

And `havingCause` in `ThrowableAssertAlternative` becomes:

```
  public ThrowableAssertAlternative<?> havingCause() {
    ThrowableAssert causeAssert = delegate.getCause();
    return new ThrowableAssertAlternative<>(causeAssert.actual);
  }
```

I return a `ThrowableAssert` since that is the type I am using. Returning `AbstractThrowableAssert<?,?>` like you propose also works, but not sure why that would be better?

Reply:
> Would it not be better to add the getCause() method in AbstractThrowableAssert on top?

Looks good to me!

> Returning AbstractThrowableAssert<?,?> like you propose also works, but not sure why that would be better?

IIRC this is a pattern used in AssertJ to allow subclasses to return a different assertion type. @joel-costigliola what do you think about this?

Reply:
yes `getCause` in `AbstractThrowableAssert` would be better as `havingCause()` looks a bit like to `hasCause(cause)` which may confuse users.

```java
assertThat(e).getCause()
             .hasMessage("cause message")
             .getRootCause()
             .hasMessage("root cause message");
```

We can return `AbstractThrowableAssert<?,?>` in case someone wants their own `MyThrowableAssert` which only need to extend `AbstractThrowableAssert<?,?>` (to be fair I don't know if people are doing that or not)

Reply:
Ok. PR updated for this now.

Reply:
@wimdeblauwe we also need to add `"getCause"` and `"getRootCause"`  to `SoftProxies.METHODS_CHANGING_THE_OBJECT_UNDER_TEST` for soft assertions to work properly.

A test must be added in `SoftAssertionsTest` like `map_soft_assertions_should_work_with_navigation_methods` but for throwable (name it `throwable_soft_assertions_should_work_with_navigation_methods`).

We want to prove in that test that after navigating to cause and root cause soft assertions work and capture properly all errors.

Reply:
Thanks for the tip. Test is added. It failed without updating `SoftProxies` and adding both methods made it work.